File: food_ordering_system/food_app_android/app/src/main/java/com/example/foodapp/MainActivity.java
package com.example.foodapp;

import android.widget.EditText;
import android.widget.Button;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.os.Message;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RatingBar;
import android.widget.Toast;
import android.Filter ;
import android.Request;
import android.Product;
import android.Shop;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText categoryEdit, priceEdit;
    private RatingBar starsRatingBar;
    private Button submitButton;
    private Handler handler;

    private final String Latitude = "37.9897";
    private final String Longitude = "23.7260";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Συνδέουμε τα components με το layout
        categoryEdit = findViewById(R.id.categoryEdit);
        priceEdit = findViewById(R.id.priceEdit);
        starsRatingBar = findViewById(R.id.starsRatingBar);
        submitButton = findViewById(R.id.submitButton);

        // Handler για να πάρουμε απάντηση από το SocketThread
        handler = new Handler(Looper.getMainLooper()) {
            @Override
            public void handleMessage(@NonNull Message msg) {//στέλνουμε την λίστα με τα φιλτραλισμένα καταστήματα στην επόμενη οθόνη
                if (msg.what == 1) {
                    ArrayList<Shop> shopList = (ArrayList<Shop>) msg.obj;
                    Intent intent = new Intent(MainActivity.this, ShopListActivity.class);
                    intent.putExtra("shopList", shopList);
                    startActivity(intent);
                } else {
                    Toast.makeText(MainActivity.this, msg.obj.toString(), Toast.LENGTH_LONG).show();
                }
            }
        };

        // οταν πατατάμε το κουμπί "search"
        submitButton.setOnClickListener(v -> {

            //παίρνουμε τις τιμές για τα φίλτρα
            String category = categoryEdit.getText().toString().trim();
            String price = priceEdit.getText().toString().trim();
            float rating = starsRatingBar.getRating();
            int stars = (int) rating;
            String starsStr = String.valueOf(stars);

            // δημιουργούμε αντικείμενο filter
            Filter filter = new Filter(category, starsStr, price, Latitude, Longitude, 1);

            // ξεκινάμε thread για αποστολή στο backend
            new SocketThread("Filter", filter, null, handler).start();
        });
    }
}

File: uni_test_application/app/src/main/java/com/example/unitestapplication/MainActivity.kt
package com.example.unitestapplication

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.example.unitestapplication.view.Subject.CreateSubject.CreateSubjectActivity
import com.example.unitestapplication.view.Subject.Monitor.ListsubjectsActivity
import com.example.unitestapplication.view.Subject.Search.SubjectSearchActivity
import com.example.unitestapplication.view.Subject.UpdateSubject.SubjectUpdateActivity
import com.example.unitestapplication.view.log_in.LogInActivity


class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val professorid = intent.getIntExtra("professorid", -1)
        Log.e("Professor id in main activity ","Professor id in main activity "+professorid);
        // Βρες το κουμπί και πρόσθεσε τη λειτουργικότητα
        val buttonCreation = findViewById<Button>(R.id.button_creation)
        buttonCreation.setOnClickListener {
            val intent = Intent(this, CreateSubjectActivity::class.java)
            intent.putExtra("professorid",professorid);
            startActivity(intent)
        }

        val buttondelete= findViewById<Button>(R.id.button_delete)
        buttondelete.setOnClickListener {
            val intent = Intent(this, SubjectSearchActivity::class.java)
            intent.putExtra("professorid",professorid);
            startActivity(intent)
        }

        // Κώδικας για την προσθήκη της λειτουργικότητας του κουμπιού "Ενημέρωση Μαθήματος"
        val buttonUpdate = findViewById<Button>(R.id.button_update)
        buttonUpdate.setOnClickListener {
            val intent = Intent(this, SubjectUpdateActivity::class.java) // Δημιουργία Intent για την δραστηριότητα ενημέρωσης
            intent.putExtra("professorid",professorid);
            startActivity(intent)  // Εκκίνηση της δραστηριότητας ενημέρωσης

        }
        val buttonmonitor = findViewById<Button>(R.id.button_success)
        buttonmonitor.setOnClickListener {
            val intent = Intent(this, ListsubjectsActivity::class.java) // Δημιουργία Intent για την δραστηριότητα ενημέρωσης
            intent.putExtra("professorid",professorid);
            startActivity(intent)  // Εκκίνηση της δραστηριότητας ενημέρωσης

        }


        val buttonexit = findViewById<Button>(R.id.button_prof_exit)
        buttonexit.setOnClickListener {
            val intent1 = Intent(this, LogInActivity::class.java) // Δημιουργία Intent για την δραστηριότητα εισόδου
            intent.putExtra("professorid",professorid);
            startActivity(intent1)  // Εκκίνηση της δραστηριότητας εισόδου

        }




    }
}



